Basics/DataInventory.ipynb

```diff
@@ -200,7 +186,7 @@
    "source": [
     "## Exploring the Data Repo with the Stack Club `Taster`\n",
     "\n",
-    "The `stackclub` library provides a `Taster` class, to explore the datasets in a given repo."
+    "The `stackclub` library provides a `Taster` class, to explore the datasets in a given repo. As an example, let's take a look at some HSC data using the `Taster`."
```

Comment:
I think this is really important. I tried to do this in the "Exploring" notebook in https://github.com/LSSTScienceCollaborations/StackClub/pull/159/commits/e0dfdc90c831f47bd33a9d457f803499ff496cc1 but I think we can do better still. BTW I wonder if the fact that there is no (longer?) a `_mapper` file in any of the folders in `/datasets/hsc/repo/rerun/DM-13666` is a problem? I think we need some input on what makes a repo a repo from @SimonKrughoff or someone.

Reply:
@kadrlica is right, it's the ability for the butler to find a `CameraMapper` that makes a repository a repository.  This means that there needs to be either a `_mapper` (old style) or `repositoryCfg.yaml` (newer style) configuration file in a directory to make it a repository.

That being said, in order for it to be useful to deal with data, there also needs to be a registry that describes the raw data somewhere in the chain of repositories.  Currently this is in a file called `registry.sqlite3`.  This file is automatically populated by the ingest phase.

FWIW, the repositories in the directory @drphilmarshall pointed to are one level further down.  E.g. `/datasets/hsc/repo/rerun/DM-13666/DEEP`.

Reply:
Thanks @SimonKrughoff . I understand from talking to @mfisherlevine at Stack Club today that `/datasets/hsc/repo` is the repo (ie, it contains a `_mapper` file or equivalent), but:

1) You said that `/datasets/hsc/repo/rerun/DM-13666/DEEP` is itself a "repository" - did you mean that?

2) When I instantiate a butler with repo path `/datasets/hsc/repo/rerun/DM-13666/DEEP` it does not fail. Merlin thinks it should, and I agree - but in v16 that is not the behavior I observe. Should we check for the existence of the files you mention, for maximum defensiveness in coding?

Reply:
Oh boy, sorry, I thought when we were talking that you were trying to point to something in between the repo root and the reruns. Your point #1 should work I think. I think there is some other instantiation syntax that uses the root and the rerun, but what you're doing is probably also fine, and if you want to restrict the data from the vast array available when using the root, to just that rerun's data, then this is probably the right thing to be doing.

But now I can't remember what problem I was trying to help you solve when telling you to point at the root. Is that way, that you say _isn't_ failing, causing problems somehow?

Reply:
Aha! I think I understand now: when a pipeline runs and puts its outputs in a re-run folder, that folder is one that a butler can understand (even though it doesn't contain a `_mapper` file - is that correct? Thinking about coding defensively, what is the test to make sure that we are in a re-run folder that the butler can read, not just any old folder?

BTW when I instantiate a butler with `/datasets/hsc/repo` as its repo, it finds zero tracts - I guess because they are all hidden away in re-run folders?

Reply:
@drphilmarshall if you take a look in `/datasets/hsc/repo/rerun/DM-13666/DEEP` you will find a file `repositoryCfg.yaml`. If you look in that file you will find:
```
RepositoryCfg_v1
_mapper: &id001 !!python/name:lsst.obs.hsc.hscMapper.HscMapper ''
_mapperArgs: {calibRoot: /datasets/hsc/repo/CALIB}
_parents:
- !RepositoryCfg_v1
  _mapper: *id001
  _mapperArgs: {calibRoot: /datasets/hsc/repo/CALIB}
  _parents: []
  _policy: null
  _root: /datasets/hsc/repo
  dirty: true
_policy: null
_root: null
dirty: true
```
This file is telling the `butler` what the `_mapper` is and where the `root` to the repo lives. This is how the `butler` can understand that this directory is part of a repo.

Reply:
@kadrlica Excellent, thanks - this is what we need. 

@rmorgan10 let's explain the above in the "Exploring" notebook, and then have the `Taster` and the Data Inventory notebook search (accurately! I think I have been mis-spelling "yaml") for `repositoryCfg.yaml` files to test for repo-hood. I'm not sure how many re-runs we should report on in the Data Inventory notebook: it'd be nice to give people good advice on which re-run they should be using by default, as well as giving them a sense of what else is there.

Reply:
Done